server/app.js

```diff
@@ -120,21 +120,30 @@ const createApp = ({
     }),
   );
 
+  // establishment toggle
+  app.use(configureEstablishment());
+
   passport.serializeUser((user, done) => done(null, user.serialize()));
   passport.deserializeUser((serializedUser, done) =>
     done(null, User.deserialize(serializedUser)),
   );
-  passport.use(
-    new AzureAdOAuth2Strategy(
-      {
-        clientID: config.auth.clientId,
-        clientSecret: config.auth.clientSecret,
-        callbackURL: config.auth.callbackUrl,
-      },
-      (accessToken, refreshToken, params, profile, done) =>
-        done(null, User.from(params.id_token)),
-    ),
-  );
+  app.use((req, res, next) => {
+    passport.use(
+      new AzureAdOAuth2Strategy(
+        {
+          clientID: config.auth.clientId,
+          clientSecret: config.auth.clientSecret,
+          callbackURL: config.auth.callbackUrl.replace(
+            'siteName',
+            req.session.establishmentName,
+          ),
```

Comment:
Are we able to get the hostname from the request, perhaps from a header passed by the load balancer? It _feels_ like doing a find/replace on the callbackUrl might catch us out later on because it's not what you'd expect when you first approach it.

It looks like we need the `callbackPath` as an environment variable, and append that to the hostname we derive automatically.

Reply:
We do that when we infer the establishment in `configureEstablishment.js - line 18`
```
    const establishmentName = pathOr('wayland', ['headers', 'host'], req)
      .split('.')[0]
      .replace(replaceUrl, '');
```
So maybe we could `req.headers.host` here?

Reply:
@LucasCairns we do? it just has a `pathOr` on it. `host` includes the full hostname not just `berwyn` etc.

Reply:
Ignore me, read the rest of your comments, hopefully the PR is better and cleaner now.